ibazel/ibazel_test.go

```diff
@@ -205,109 +220,64 @@ func TestIBazelRun_firstPass(t *testing.T) {
 	}
 
 	mockBazel.AssertActions(t, expected)
+}
 
-	if cmd.Stdout != os.Stdout {
-		t.Errorf("Didn't set Stdout correctly")
-	}
-	if cmd.Stderr != os.Stderr {
-		t.Errorf("Didn't set Stderr correctly")
-	}
-	if cmd.SysProcAttr.Setpgid != true {
-		t.Errorf("Never set PGID (will prevent killing process trees -- see notes in ibazel.go")
-	}
-
-	if err := cmd.Wait(); err != nil {
-		t.Errorf("Subprocess was never started. State: %v, Err: %v", cmd.ProcessState, err)
+func TestIBazelRun_notifyPrexistiingJobWhenStarting(t *testing.T) {
+	oldDefaultCommand := commandDefaultCommand
+	commandDefaultCommand = func(b bazel.Bazel, target string, args []string) command.Command {
+		// Don't do anything
+		return &mockCommand{
+			b:      b,
+			target: target,
+			args:   args,
+		}
 	}
-}
+	defer func() { commandDefaultCommand = oldDefaultCommand }()
 
-func TestIBazelRun_killPrexistiingJobWhenStarting(t *testing.T) {
 	i, err := New()
 	if err != nil {
 		t.Errorf("Error creating IBazel: %s", err)
 	}
-
 	defer i.Cleanup()
 
-	// Create a process that has been started and can be killed
-	toKill := exec.Command("sleep", "5s")
-	toKill.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
-	i.cmd = toKill
-	toKill.Start()
+	i.args = []string{"--do_it"}
 
-	cmd := exec.Command("ls")
-	execCommand = func(name string, arg ...string) *exec.Cmd {
-		return cmd
+	cmd := &mockCommand{
+		notifiedOfChanges: false,
 	}
+	i.cmd = cmd
 
-	i.run("//path/to:target")
+	path := "//path/to:target"
+	i.run(path)
+
+	if !cmd.notifiedOfChanges {
+		t.Errorf("The preiously running command was not notified of changes")
+	}
 
 	expected := [][]string{
 		[]string{"Cancel"},
 		[]string{"WriteToStderr"},
 		[]string{"WriteToStdout"},
-		[]string{"Run", "--script_path=.*", "//path/to:target"},
+		// it's last action was to call cmd.start, but that was mocked out,
```

Comment:
We should make sure that cmd.start was only called once. That was a bug I noticed when I was taking a stab at changing this test

Reply:
Should already be an error
https://golang.org/src/os/exec/exec.go?s=9387:9414#L338